File: my_app/lib/roundEligibilityService.dart
/*
*Ella Muro
*8 June 2026
*Round Eligibility Logic
*/

class RoundEligibilityService {

  //Fields of the class
  final int totalRounds;
  final int totalPrepicks;

  //Constructor
  const RoundEligibilityService({
    this.totalRounds = 9,
    this.totalPrepicks = 2,
  });

  //True/false list for rounds (index 0 = Round 1)
  List<bool> computeRoundEligibility(int weeksAllowed) {
    return List.generate(
      totalRounds,
      (i) => (i + 1) <= weeksAllowed,
    );
  }

  //True/false list for prepicks (index 0 = Prepick 1)
  List<bool> computePrepickEligibility(int prepicksAllowed) {
    return List.generate(
      totalPrepicks,
      (i) => (i + 1) <= prepicksAllowed,
    );
  }
}

File: backend/services/roundEligibilityService.js
/*
* Ella Muro
* 8 June 2026
* Round Eligibility Logic
*/

class RoundEligibilityService {

    constructor() {

        this.totalRounds = 9;
        this.totalPrepicks = 2;

    }

    // Returns a true/false array for rounds
    // Index 0 = Round 1
    computeRoundEligibility(weeksAllowed) {

        return Array.from(

            { length: this.totalRounds },

            (_, i) => (i + 1) <= weeksAllowed

        );

    }

    // Returns a true/false array for prepicks
    // Index 0 = Prepick 1
    computePrepickEligibility(prepicksAllowed) {

        return Array.from(

            { length: this.totalPrepicks },

            (_, i) => (i + 1) <= prepicksAllowed

        );

    }

}

module.exports = RoundEligibilityService;